tests/ui-toml/undocumented_unsafe_blocks/undocumented_unsafe_blocks.rs

```diff
@@ -0,0 +1,24 @@
+#![deny(clippy::undocumented_unsafe_blocks)]
+#![allow(clippy::missing_safety_doc)]
+
+fn main() {
+    // Safety: A safety comment
+    let _some_variable_with_a_very_long_name_to_break_the_line =
+        unsafe { a_function_with_a_very_long_name_to_break_the_line() };
+
+    // Safety: Another safety comment
+    const _SOME_CONST_WITH_A_VERY_LONG_NAME_TO_BREAK_THE_LINE: u32 =
+        unsafe { a_const_function_with_a_very_long_name_to_break_the_line() };
+
+    // Safety: Yet another safety comment
+    static _SOME_STATIC_WITH_A_VERY_LONG_NAME_TO_BREAK_THE_LINE: u32 =
+        unsafe { a_const_function_with_a_very_long_name_to_break_the_line() };
+}
+
+pub unsafe fn a_function_with_a_very_long_name_to_break_the_line() -> u32 {
+    1
+}
+
+pub const unsafe fn a_const_function_with_a_very_long_name_to_break_the_line() -> u32 {
+    2
+}
```

Comment:
I think, just in case, this file should be the same as `tests/ui/undocumented_unsafe_blocks.rs` (minus the safety comments). This way if something else breaks with this configuration option we can see.

Reply:
Ok! I understand the idea of having all the same tests with this configuration and this makes sense for me. But I could not understand why `minus the safety comments`. The configuration will affect just a small subset of all tests. So, in summary, we will have almost the same .stderr file. The only difference would be those cases that we already have here at `tests/ui-toml/toml_unknown_key/conf_unknown_key.stderr` that don't lint. But we would have to have the safety comments in all test cases to test is properly, don't we?

Maybe I am just not undestanding your idea.

Reply:
> minus the safety comments

Sorry, I meant that in regard to `should be the same`; For the ones affected by the configuration option, they would be of course different :D

Reply:
Ok, great! I will do that

Reply:
Done: https://github.com/rust-lang/rust-clippy/pull/10886/commits/928c712c0d3a7b9766f38046a573295c11767933
Two interesting things:
- I had to duplicate code and create a new `auxiliary/proc_macro_unsafe.rs` file under the test folder. I don't know if there is another way to use auxiliary functions in ui-toml tests without this duplication
- And there was another test case that don't fire with this configuration. It is the unsafe block at line 498. And it makes sense to don't lint this case with the configuration. It was cool to see that but I was not expecting it :)